### PR TITLE
Add check for invalid domain and throw better exception

### PR DESCRIPTION
**Pull request by nikolaseu**

### CODE REVIEW

File: auth0/src/main/java/com/auth0/android/Auth0.java [hunk 1]
```diff
@@ -139,6 +139,9 @@ private String resolveConfiguration(String configurationDomain, String domainUrl
         String url = ensureUrlString(configurationDomain);
         if (configurationDomain == null && domainUrl != null) {
             final HttpUrl domainUri = HttpUrl.parse(domainUrl);
+            if (domainUri == null) {
```
Comment (nikolaseu) [inline auth0/src/main/java/com/auth0/android/Auth0.java]: maybe it's better to check at ensureUrlString? also here the domainUrl was already modified by ensureUrlString called at the constructor

Comment (nikolaseu) [inline auth0/src/main/java/com/auth0/android/Auth0.java]: also we can parse only once and save the HttpUrl directly?

Comment (lbalmaceda) [inline auth0/src/main/java/com/auth0/android/Auth0.java]: Let's use `HttpUrl` for domainUrl and configurationUrl

**Comment by nikolaseu** [comment]
@lbalmaceda updated

**Comment by lbalmaceda** [comment]
LGTM

**Pull request merged**